File: CS/MvcThrowCustomExceptionDashboardErrorToast/Controllers/CustomDashboardController.cs
using System.Net;
using System.Web.Mvc;
using DevExpress.DashboardWeb.Mvc;

namespace MvcThrowCustomExceptionDashboardErrorToast.Controllers {
    public class CustomDashboardController : DashboardController {
        protected override void OnException(ExceptionContext context) {
            var exception = context.Exception;
            if(exception != null && context.HttpContext != null) {
                var response = context.HttpContext.Response;
                response.StatusCode = (int)HttpStatusCode.BadRequest;
                response.ContentType = "application/json";

                CustomException customException = exception as CustomException;
                bool isCustomErrorsEnabled = System.Web.HttpContext.Current != null ? System.Web.HttpContext.Current.IsCustomErrorEnabled : true;
                string message = customException != null ? (isCustomErrorsEnabled ? CustomException.SafeMessage : CustomException.UnsafeMessage) : "";
                response.Write(GetJson(message));

                context.ExceptionHandled = true;
            }
        }

        static string GetJson(string message) {
            return $"{{ \"Message\":\"{message}\" }}";
        }
    }
}

File: VB/MvcThrowCustomExceptionDashboardErrorToast/Controllers/CustomDashboardController.vb
Imports Microsoft.VisualBasic
Imports System.Net
Imports System.Web.Mvc
Imports DevExpress.DashboardWeb.Mvc

Namespace MvcThrowCustomExceptionDashboardErrorToast.Controllers
	Public Class CustomDashboardController
		Inherits DashboardController
		Protected Overrides Sub OnException(ByVal context As ExceptionContext)
			Dim exception = context.Exception
			If exception IsNot Nothing AndAlso context.HttpContext IsNot Nothing Then
				Dim response = context.HttpContext.Response
				response.StatusCode = CInt(Fix(HttpStatusCode.BadRequest))
				response.ContentType = "application/json"

				Dim customException As CustomException = TryCast(exception, CustomException)
				Dim isCustomErrorsEnabled As Boolean = If(System.Web.HttpContext.Current IsNot Nothing, System.Web.HttpContext.Current.IsCustomErrorEnabled, True)
				Dim message As String = If(customException IsNot Nothing, (If(isCustomErrorsEnabled, CustomException.SafeMessage, CustomException.UnsafeMessage)), "")
				response.Write(GetJson(message))

				context.ExceptionHandled = True
			End If
		End Sub

		Private Shared Function GetJson(ByVal message As String) As String
			Return $"{{ ""Message"":""{message}"" }}"
		End Function
	End Class
End Namespace